File: data/EXP01/golang_no_system_var/2358/2358.output_3.go
func maximumGroups(grades []int) int {
    sort.Ints(grades)
    numGroups := 0
    currSum := 0
    currCount := 0

    for i := 0; i < len(grades); i++ {
        currSum += grades[i]
        currCount++

        if i < len(grades)-1 && currSum >= grades[i+1] {
            numGroups++
            currSum = 0
            currCount = 0
        }
    }

    if currCount > 0 {
        numGroups++
    }

    return numGroups
}

File: data/EXP01/python/2358/2358.output_3.py
# @lc app=leetcode id=2358 lang=python3
from typing import List

class Solution:
    def maximumGroups(self, grades: List[int]) -> int:
        grades.sort(reverse=True)  # Sort the grades in descending order
        groups = []
        current_group_sum = 0
        current_group_count = 0
        
        for grade in grades:
            current_group_sum += grade
            current_group_count += 1
            
            if len(groups) == 0 or current_group_sum < sum(groups[-1]):
                groups.append([grade])
                current_group_sum = grade
                current_group_count = 1
            else:
                groups[-1].append(grade)
        
        return len(groups)